### PR TITLE
Try to address an unhandled exception that's crashing instances

### PR DESCRIPTION
**Pull request by jeremy-rifkin**

### CODE REVIEW

File: lib/handlers/compile.ts [hunk 1]
```diff
@@ -144,15 +144,28 @@ export class CompileHandler {
...
-            if (bodyData) {
-                proxyReq.setHeader('Content-Length', Buffer.byteLength(bodyData));
-                proxyReq.write(bodyData);
+            try {
+                if (bodyData) {
+                    proxyReq.setHeader('Content-Length', Buffer.byteLength(bodyData));
+                    proxyReq.write(bodyData);
+                }
+            } catch (e: any) {
+                Sentry.captureException(e);
```
Comment (jeremy-rifkin) [inline lib/handlers/compile.ts:162]: Maybe very overkill

File: lib/handlers/compile.ts [hunk 2]
```diff
@@ -144,15 +144,28 @@ export class CompileHandler {
...
             if (contentType === 'application/json') {
                 bodyData = JSON.stringify(body);
-            }
-
-            if (contentType === 'application/x-www-form-urlencoded') {
+            } else if (contentType === 'application/x-www-form-urlencoded') {
                 bodyData = body;
+            } else {
+                Sentry.captureException(
+                    new Error(`Unexpected Content-Type received by /compiler/:compiler/compile: ${contentType}`),
```
Comment (jeremy-rifkin) [inline lib/handlers/compile.ts:151]: I'm pretty sure whatever is leading to an extraneous `bodyData` must be a Content-Type we aren't expecting

**Comment by jeremy-rifkin** [comment]
Fixes at least part of #4516

File: lib/handlers/compile.ts (hunk 1 above)
Comment (mattgodbolt) [reply lib/handlers/compile.ts:162]: if it's waking me up in the night sending me messages cos the site's crashing, it's not overkill :D

**Review by mattgodbolt — APPROVED** [review]
Thanks so much.

File: lib/handlers/compile.ts (hunk 2 above)
Comment (mattgodbolt) [reply lib/handlers/compile.ts:151]: Possibly, though in local testing with a variety of ill-formed and well-formed-but-wrong-type I couldn't get it to trigger.

**Pull request merged**